src/rulesets/rest/2022-05-25/json-api-rules/resource-object-rules.ts

```diff
@@ -17,11 +17,13 @@ const responseDataForPatch = new ResponseBodyRule({
   rule: (responseAssertions) => {
     responseAssertions.body.added.matches({
       schema: {
+        type: 'object',
```

Comment:
This is slightly better than what we had before, but what we had wasn't very good. This should be asserting a [valid JSON API PATCH response](https://jsonapi.org/format/#crud-updating-responses) for a 200, similar to how it's done below for GET and POST responses (see `dataProperty` below).

Reply:
I wonder if we should refactor a bit -- a rule requiring a JSON API resource response could use a match function that looks at method, response code and content type, to consolidate the decision logic on when a response should contain a resource (200 or 201 for a POST, 200 for GET and PATCH, etc.)